Framework/src/InfrastructureGenerator.cxx

```diff
@@ -800,4 +808,48 @@ void InfrastructureGenerator::generatePostProcessing(WorkflowSpec& workflow, con
   }
 }
 
+template <typename Type>
+auto createSinkInput(const std::string& detectorName, const std::string& name) -> framework::InputSpec
+{
+  const auto outputSpec = Type::createOutputSpec(detectorName, name);
+  auto input = DataSpecUtils::matchingInput(outputSpec);
+  input.binding = name;
+  return input;
+}
+
+void InfrastructureGenerator::generateBookkeepingQualitySink(WorkflowSpec& workflow, const InfrastructureSpec& infrastructureSpec)
+{
+  framework::Inputs sinkInputs{};
+
+  for (const auto& checkSpec : infrastructureSpec.checks) {
+    if (checkSpec.active && checkSpec.exportToBookkeeping) {
+      ILOG(Debug, Support) << "Adding input to BookkeepingSink from check " << checkSpec.checkName << " and detector: " << checkSpec.detectorName << ENDM;
+      sinkInputs.emplace_back(createSinkInput<Check>(checkSpec.detectorName, checkSpec.checkName));
+    }
+  }
+
+  for (const auto& aggregatorSpec : infrastructureSpec.aggregators) {
+    if (aggregatorSpec.active && aggregatorSpec.exportToBookkeeping) {
+      ILOG(Debug, Support) << "Adding input to BookkeepingSink from aggregator " << aggregatorSpec.aggregatorName << " and detector: " << aggregatorSpec.detectorName << ENDM;
+      sinkInputs.emplace_back(createSinkInput<Aggregator>(aggregatorSpec.detectorName, aggregatorSpec.aggregatorName));
+    }
+  }
+
+  if (sinkInputs.empty()) {
+    ILOG(Debug, Support) << "BookkeepingSink is not being created because there any suitable inputs with quality objects were not found" << ENDM;
```

Comment:
I don't quite understand the part `because there any suitable input`, could you rephrase please?